### PR TITLE
Super class expression parenthesis

### PR DESCRIPTION
**Pull request by sanex3339**
Fixes https://github.com/javascript-obfuscator/javascript-obfuscator/issues/326

### CODE REVIEW

**Comment by sanex3339** [comment]
@michaelficarra need your review

**Comment by sanex3339** [comment]
@michaelficarra ping

File: escodegen.js [hunk 1]
```diff
@@ -1145,7 +1145,17 @@
                 result = join(result, this.generateExpression(stmt.id, Precedence.Sequence, E_TTT));
             }
             if (stmt.superClass) {
-                fragment = join('extends', this.generateExpression(stmt.superClass, Precedence.Assignment, E_TTT));
+                var superClassExpression = this.generateExpression(stmt.superClass, Precedence.Assignment, E_TTT);
```
Comment (michaelficarra) [inline escodegen.js]: None of the rest of this is necessary. [ClassHeritage](https://tc39.es/ecma262/#prod-ClassHeritage) is a LeftHandSideExpression, not an AssignmentExpression. We should just replace `Precedence.Assignment` on this line with `Precedence.Unary`.

Comment (sanex3339) [reply escodegen.js]: Will try

Comment (sanex3339) [reply escodegen.js]: It works, thank you

**Comment by sanex3339** [comment]
@michaelficarra fixed

**Pull request merged**